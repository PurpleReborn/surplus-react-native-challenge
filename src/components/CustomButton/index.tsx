import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import React from 'react';
import CustomText from '../CustomText';
import colors from '../../themes/colors';

const CustomButton = ({
  text,
  bgColor,
  mb,
  mt,
  disabled,
  onPress,
}: {
  text: string;
  bgColor?: string;
  mb?: number;
  mt?: number;
  disabled?: boolean;
  onPress?: () => void;
}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={disabled}
      style={[
        {
          backgroundColor: disabled
            ? colors.greenDisabled
            : bgColor
            ? bgColor
            : colors.green,
          marginBottom: mb ? mb : 0,
          marginTop: mt ? mt : 0,
        },
        styles.wrapper,
      ]}>
      <CustomText
        text={text}
        style={{fontWeight: '500', textAlign: 'center'}}
        color="white"
      />
    </TouchableOpacity>
  );
};

export default CustomButton;

const styles = StyleSheet.create({
  wrapper: {
    padding: 12,
    borderRadius: 24,
  },
});
