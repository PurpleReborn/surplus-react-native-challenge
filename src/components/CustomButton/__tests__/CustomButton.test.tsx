import React from 'react';
import TestRenderer from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';
import CustomButton from '..';

describe('CustomButton', () => {
  const onPressMock = jest.fn();
  const text = 'Press me!';
  const tree = TestRenderer.create(
    <CustomButton text={text} onPress={onPressMock} />,
  );

  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('contains a TouchableOpacity', () => {
    const touchableNodes = tree.root.findAllByType(TouchableOpacity);
    expect(touchableNodes).toHaveLength(1);
  });

  it('disables TouchableOpacity when disabled prop is true', () => {
    const disabledTree = TestRenderer.create(
      <CustomButton text={text} disabled={true} onPress={onPressMock} />,
    );
    const touchableNodes = disabledTree.root.findAllByType(TouchableOpacity);
    expect(touchableNodes[0].props.disabled).toBe(true);
  });

  it('calls onPress function when TouchableOpacity is pressed', () => {
    const touchableNodes = tree.root.findAllByType(TouchableOpacity);
    touchableNodes[0].props.onPress();
    expect(onPressMock).toHaveBeenCalled();
  });
});
