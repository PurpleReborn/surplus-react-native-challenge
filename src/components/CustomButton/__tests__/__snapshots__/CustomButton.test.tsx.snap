// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#51B8A7",
      "borderRadius": 24,
      "marginBottom": 0,
      "marginTop": 0,
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "500",
        "marginBottom": 0,
        "marginTop": 0,
        "textAlign": "center",
      }
    }
  >
    Press me!
  </Text>
</View>
`;
